libethereum/ExtVM.cpp

```diff
@@ -95,30 +94,62 @@ void go(unsigned _depth, Executive& _e, OnOpFunc const& _onOp)
 
 bool ExtVM::call(CallParameters& _p)
 {
-	Executive e(m_s, envInfo(), m_sealEngine, depth + 1);
+	Executive e{m_s, envInfo(), m_sealEngine, depth + 1};
 	if (!e.call(_p, gasPrice, origin))
 	{
 		go(depth, e, _p.onOp);
 		e.accrueSubState(sub);
 	}
 	_p.gas = e.gas();
 
-	return !e.excepted();
+	if (e.excepted())
+		return false;
+
+	// FIXME: Make sure move constructor is really used.
+	m_orig.children.emplace_back(e.takeSnapshot());
```

Comment:
The name of `takeSnapshot` would suggest to store the current state of the sub-call, but I think it stores the previous state, right? Perhaps we should call it something like `reversionLog`?
Also I think `m_orig`, `takeSnapshot` and `AccountSnapshot` should all have similar names.